src/webgpu/api/operation/resource_init/buffer.spec.ts

```diff
@@ -5,7 +5,276 @@ export const description = `
 Test uninitialized buffers are initialized to zero when read
 (or read-written, e.g. with depth write or atomics).
 
-TODO
+Here lists all the buffer usages to test:
+- in writeBuffer()
+- in map async (read and partial write)
+- created with mappedAtCreation === true
+- as copy source (TODO)
+- as copy destination in a partial copy (TODO)
+- in ResolveQuerySet() (TODO)
+- as uniform / read-only storage / storage buffer (TODO)
+- as vertex / index buffe (TODO)
+- as indirect buffer (TODO)
 `;
 
 export const g = makeTestGroup(GPUTest);
+
+g.test('partial_write_buffer')
+  .paramsSubcasesOnly(u => u.combine('offset', [0, 8, -12]))
```

Comment:
nit: all tests should have a .desc() so that we can use the CTS as a test plan viewer.